### ISSUE TITLE
requestDevice returnes empty

### ISSUE THREAD

**Issue by Richard-Mathie** [thegecko/webusb#7]
I can get a device with `node-usb` but `webusb` throws an error _no devices found_

## with `webusb`:
```javascript
var usb = require("usb").usb;
usb.requestDevice({filters: [{vendorId: 0x0fcf}]}).then(console.log).catch(console.log)
```
gives:
```
Error: requestDevice error: no devices found
    at adapter_1.adapter.listUSBDevices.then.devices (...../webusb/lib/usb.js:159:35)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
> 
```
## with `node-usb`:
```javascript
var usb = require("usb");
usb.getDeviceList().filter(d => {return d.deviceDescriptor.idVendor == 0x0fcf})
```
output:
```javscript
[ Device {
    busNumber: 1,
    deviceAddress: 7,
    deviceDescriptor: 
     { bLength: 18,
       bDescriptorType: 1,
       bcdUSB: 512,
       bDeviceClass: 0,
       bDeviceSubClass: 0,
       bDeviceProtocol: 0,
       bMaxPacketSize0: 32,
       idVendor: 4047,
       idProduct: 4105,
       bcdDevice: 256,
       iManufacturer: 1,
       iProduct: 2,
       iSerialNumber: 3,
       bNumConfigurations: 1 },
    portNumbers: [ 1 ] } ]
```

## in chrome WebUSB works
```javascript
usb.requestDevice({filters: [{vendorId: 0x0fcf}]}).then(console.log).catch(console.log)
> Promise {<pending>}
> USBDevice {usbVersionMajor: 2, usbVersionMinor: 0, usbVersionSubminor: 0, deviceClass: 0, deviceSubclass: 0, …}
```

I guess it's being dropped by the filters somehow, but I am struggling to work out where switching between the different adapters js/ts and the promise api's. Can you point us into the right direction?

**Comment by thegecko** [comment]
Hi @Richard-Mathie The WebUSB specification highlights that WebUSB enabled devices need to have a WebUSB capability descriptor. Therefore this library will only return devices which satisfy that constraint.

The part of the specification outlining this can be seen here:
https://wicg.github.io/webusb/#webusb-platform-capability-descriptor

And this is where I filter based on that capability:
https://github.com/thegecko/webusb/blob/master/src/adapter.ts#L259

**Comment by thegecko** [comment]
It's also possible it doesn't return WebUSB devices without URLs (I haven't had a chance to test that) so if that's your case, please let me know.

**Comment by Richard-Mathie** [comment]
Yeah I had a look through, and it might have been the URL filter, its a bit hard to trace for me.

**Comment by thegecko** [comment]
You could try disabling each of the filters/checks in turn in the code I point to above and recompiling to see progress?

**Comment by thegecko** [comment]
I've had a chance to test this now and there were errors when a WebUSB device didn't have a URL. Now fixed in #11 and I'll make a release...

**Comment by thegecko** [comment]
@Richard-Mathie  Did this fix your issue?

**Comment by thegecko** [comment]
Closing due to inactivity

**Comment by thegecko** [comment]
@Richard-Mathie It seems Chrome allows the user to filter against all devices (not just those with the WebUSB characteristic). Therefore I've removed this restriction in PR #13 (and released in 1.0.4) which may have been the cause of your issue.
